### PR TITLE
Create "Solving SAT problems with Grover" kata

### PR DESCRIPTION
**Pull request by tcNickolas**
Includes two variants of SAT problems, the canonical one and the Exactly-1 3-SAT problem.

### CODE REVIEW

**Review by ScottCarda-MS — APPROVED** [review]

File: katas/content/solving_sat/index.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+@[example]({"id": "solving_sat__e2edemo", "codePath": "./examples/SolvingSATWithGroverDemo.qs"})
+
+
+@[section]({
+    "id": "solving_sat__conclusion",
+    "title": "Conclusion"
+})
+
+Congratulations! In this kata you learned to solve Boolean satisfiability problems using Grover's search.
```
Comment (cesarzc) [inline katas/content/solving_sat/index.md:116]:
```suggestion
Congratulations! In this kata you learned to solve the Boolean satisfiability problems using Grover's search.
```

File: katas/content/solving_sat/index.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+- Controlled gates
+- Oracles, in particular marking oracles
+- Grover's search algorithm
+
+@[section]({
+    "id": "solving_sat__cnf",
+    "title": "Canonical Satisfiability Problem"
+})
+
+Boolean satisfiability problem is the problem of determining whether there exists an assignment of Boolean variables $x_j$
```
Comment (cesarzc) [inline katas/content/solving_sat/index.md:31]:
```suggestion
The Boolean satisfiability problem is the problem of determining whether there exists an assignment of Boolean variables $x_j$
```

File: katas/content/solving_sat/index.md (hunk 1 above)
Comment (tcNickolas) [reply katas/content/solving_sat/index.md:116]: Are you sure about that? I asked AI and it says this one shouldn't have "the"...

Comment (cesarzc) [reply katas/content/solving_sat/index.md:116]: I didn't read this one correctly, I read it as problem instead of problems. Closing this comment.

**Review by cesarzc — APPROVED** [review]

**Pull request merged**